jarviscli/plugins/qr_generator.py

```diff
@@ -0,0 +1,168 @@
+
+from plugin import plugin,require
+from colorama import Fore
+import requests
+import os
+import re
+
+
+@require(network=True)
+@plugin("qr")
+
+class QRGenerator:
+    """
+    Provides you with a png image of a QR for a specified URL
+
+    Enter 'qr' or 'QR' to use:
+    
+
+    Attribution:
+        QR generated from https://goqr.me/api/
+    """
+    
+    def __init__(self):
+        self.url = "http://api.qrserver.com/v1/create-qr-code/"
+    
+
+    def __call__(self,jarvis,s):
+        if s == 'help':
+            self._print_help(jarvis)
+            return
+
+        jarvis.say("\ntype 'qr  help' for additional information")
+        jarvis.say("")
+
+        prompt_url = ("\nType the URL that you want to create a QR code \n" 
+                      "or 'end' if you wish to exit: ")
+        # get url from user prompt
+        url = self._get_url(prompt_url,jarvis)
+        if url == "end":
+            return
+        jarvis.say("")
+
+        prompt_path = ("\nType the filepath where you want the QR to be stored \n"
+                       "or 'end' if you wish to exit: ")
+        # get path from user prompt
+        self.path = self._get_path(prompt_path,jarvis)
+        if self.path == "end":
+            return
+        self.path = self.path + "\\"
```

Comment:
Does not work on Linux.
Better use:

os.path.join(self.path, self.filename + '.png') 

in line 130